### DISCUSSION TITLE
clonning error

### DISCUSSION THREAD

**Discussion by Alfredo-Junqueira-75** [robolectric/robolectric#8935] [Q&A] [answered]
hi everyone... Can someone help me? I'm trying to clone the repository to my machine but it's giving me an error

![image](https://github.com/robolectric/robolectric/assets/103441772/38da4517-62af-4b78-b826-9952b70ac642)

**Comment by utzcoz** [comment, marked as answer]
The clone command is simple:

```
git clone https://github.com/robolectric/robolectric.git
```

If it still has issues, please check your network and search the failed messages on the internet.

**Reply by Alfredo-Junqueira-75** [reply]
Thanks for help, I already found the answer, it was related to the git buffer size.

**Reply by utzcoz** [reply]
Glad to know you have fixed it.

**Comment by harshal-rembhotkar** [comment]
Hey @Alfredo-Junqueira-75 ,it is not a cloning error , it's just happened when your internet connection is OFF or  loosed.

**Reply by Alfredo-Junqueira-75** [reply]
That's what it seemed like. I checked my internet connection, and everything was fine. I waited for a few hours, but every time I tried, the same error occurred. However, I managed to solve the problem by increasing the buffer size used by Git to handle HTTP data during operations like clone, push, and pull. This allowed Git to handle a larger amount of data at once.

https://stackoverflow.com/a/61920529/21881999

**Reply by harshal-rembhotkar** [reply]
Glad to know you have solved it.

**Reply by Alfredo-Junqueira-75** [reply]
Yes, it's a relief to have it sorted out. Thanks for your support!

**Comment by hoisie** [comment]
Alternatively, you could clone the repo over ssh, which may be more reliable. If you are planning to push code to a repository, it is better to use ssh for remotes.

**Reply by Alfredo-Junqueira-75** [reply]
Thanks for the suggestion! I'll give cloning over SSH a try, and I agree, using SSH for pushing seems like the way to go.
